### DISCUSSION TITLE
name 'delete' is not defined

### DISCUSSION THREAD

**Discussion by BackSeat** [sqlalchemy/sqlalchemy#8365] [Usage Questions] [answered]
I can't help feeling that this is a user error, but... The [documentation](https://docs.sqlalchemy.org/en/20/tutorial/orm_data_manipulation.html?highlight=orm+v2#orm-enabled-delete-statements) implies that this should work:

```
session.execute(
            delete(PlaylistRows)
            .where(PlaylistRows.id.in_(ids))
        )
```
I get: `<class 'NameError'>: name 'delete' is not defined`

What am I doing wrong?

**Comment by gordthompson** [comment, marked as answer]
Have you imported the name?

```python
from sqlalchemy import delete
```

**Reply by BackSeat** [reply]
Thank you, that (of course) fixed it. I feel a bit stupid...in my defence, the page I referenced doesn't contain the word 'import' anywhere. Perhaps it should mention, somewhere, that it's necessary to import delete.
